ui/perfherder/compare/CompareTable.jsx

```diff
@@ -28,24 +44,212 @@ export default class CompareTable extends React.PureComponent {
+      if (shouldBeNumber) {
+        // eslint-disable-next-line no-restricted-globals
+        if (isNaN(a[key]) && !Array.isArray(a[key])) a[key] = 0;
+        // eslint-disable-next-line no-restricted-globals
+        if (isNaN(b[key]) && !Array.isArray(b[key])) b[key] = 0;
+      } else {
+        a[key] = typeof a[key] === 'undefined' ? '' : a[key];
+        b[key] = typeof b[key] === 'undefined' ? '' : b[key];
+      }
+      const ak = Array.isArray(a[key]) ? a[key].length : a[key];
+      const bk = Array.isArray(b[key]) ? b[key].length : b[key];
+      if (ak < bk) {
+        if (sortOrderAsc) return -1;
+        return 1;
+      }
+      if (ak > bk) {
+        if (sortOrderAsc) return 1;
+        return -1;
+      }
+      return 0;
+    };
+  };
+
+  sortBy = (key, data) => {
+    const { sortOrderAsc } = this.state;
+    let {
+      borderTopColor,
+      borderTopWidth,
+      borderBottomColor,
+      borderBottomWidth,
+    } = this.state;
+    const columnName = key;
+    // every other columns except "name" are numbers
+    const shouldBeNumber = key !== 'name';
+    data.sort(this.compareBy(key, shouldBeNumber));
+    const sorted = true;
+    if (sorted) {
+      if (sortOrderAsc) {
+        borderTopColor = '#333';
+        borderTopWidth = '1px';
+        borderBottomColor = '';
+        borderBottomWidth = '';
+        this.setState({
+          columnName,
+          sortOrderAsc: !sortOrderAsc,
+          sorted,
+          borderTopColor,
+          borderTopWidth,
+          borderBottomColor,
+          borderBottomWidth,
+        });
+      } else {
+        borderBottomColor = '#333';
+        borderBottomWidth = '1px';
+        borderTopColor = '';
+        borderTopWidth = '';
+        this.setState({
+          columnName,
+          sortOrderAsc: !sortOrderAsc,
+          sorted,
+          borderTopColor,
+          borderTopWidth,
+          borderBottomColor,
+          borderBottomWidth,
+        });
+      }
+    }
+    // console.log(`sortOrderAsc: ${sortOrderAsc}`);
+    // console.log(`sortBy sorted: ${sorted}`);
+
+    this.forceUpdate();
+  };
+
   render() {
     const { data, testName } = this.props;
+    const {
+      columnName,
+      sorted,
+      borderTopColor,
+      borderTopWidth,
+      borderBottomColor,
+      borderBottomWidth,
+    } = this.state;
+    // console.log(`data: ${JSON.stringify(data)}`);
+    // console.log(`render sorted: ${JSON.stringify(sorted)}`);
     return (
       <Table sz="small" className="compare-table mb-0 px-0" key={testName}>
         <thead>
-          <tr className="subtest-header bg-lightgray">
-            <th className="text-left">
-              <span>{testName}</span>
+          <tr
+            className={`${
+              sorted ? 'subtest-header-visible ' : 'subtest-header '
+            }bg-lightgray`}
+          >
+            <th
+              style={{
+                borderTopColor: columnName === 'name' ? borderTopColor : '',
```

Comment:
This style thing repeats the same 4 properties for all the sortable headings. Not sure how I can reduce this redundancy without major changes.

Reply:
We've moved away from using inline styles so I'd rather you find a way to toggle between the different styles based on default, asc or desc by using class names (so CSS in the perf style sheet).

Reply:
I'm thinking that this code will survive only until you finish the refactoring from angular to react that why I didn't chose to invest too much time on this.
So far I didn't find a more decent solution other than those inline styles, but I am going to dig more. Thanks!